Assets/MixedRealityToolkit-SDK/Features/Diagnostics/MemoryUseTracker.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Diagnostics;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.DiagnosticsSystem
+{
+    internal class MemoryUseTracker
+    {
+        private Process currentProcess = Process.GetCurrentProcess();
+        private MemoryReading[] memoryReadings = new MemoryReading[10];
+        private int index = 0;
+
+        public MemoryReading GetReading()
+        {
+            var reading = new MemoryReading()
+            {
+                VirtualMemoryInBytes = currentProcess.VirtualMemorySize64,
+                WorkingSetMemoryInBytes = currentProcess.WorkingSet64,
+                GCMemoryInBytes = GC.GetTotalMemory(false)
+            };
+
+            memoryReadings[index] = reading;
+            index = (index + 1) % memoryReadings.Length;
+
+            var sum = memoryReadings.Aggregate(new MemoryReading(), (a, b) => a + b);
+            return sum / memoryReadings.Length;
+        }
+
+        public struct MemoryReading
```

Comment:
internal?

Reply:
Doesn't need to be because it's encapsulated by the MemoryUseTrackerClass